File: Swift-C++-IR-Comparison/test/conditionalLiveness/swift/classInitVsAssign.swift
// example taken from "Swift's High-Level IR: A Case Study"

class SomeClass {
  var text: String
  init(text: String) {
    self.text = text
  }
  func ask() {
    print(text)
  }
}

func testDI(cond: Bool) {
  var c: SomeClass
  if cond {
    c = SomeClass(text: "12345")
  }

  c = SomeClass(text: "6789")  // init or assign?

  c.ask()

}

testDI(cond: Int.random(in: 1..<100) > 50)

File: AutoTranspiledTests/try-2/classInitVsAssign.cpp
 

#include <iostream>

class SomeClass {
public:
	std::string text;
	SomeClass(std::string text) {
		this->text = text;
	}

	void ask() {
		std::cout << text << std::endl;
	}
};

void testDI(bool cond) {
	SomeClass *c;
	if (cond) {
		c = new SomeClass("12345");
	}

	c = new SomeClass("6789");  // init or assign?

	c->ask();

}

int main() {
	testDI((rand() % 100) > 50);
	return 0;
}